### DISCUSSION TITLE
"Pretest" socket address?

### DISCUSSION THREAD

**Discussion by Exergist** [jchristn/SuperSimpleTcp#49] [Q&A] [answered]
Is there functionality available to "pretest" or check a socket address (IP:port) prior to executing a `Connect` action? 

Sure I can handle the `Only one usage of each socket address (protocol/network address/port) is normally permitted` exception if the socket address is currently occupied, but I figured I'd ask to see if there might be a better way. Thanks!

**Comment by jchristn** [comment, marked as answer]
Hi @Exergist not natively within the library.  I did find this which may be useful for you: https://softwarebydefault.com/2013/02/22/port-in-use/ and also https://stackoverflow.com/questions/570098/in-c-how-to-check-if-a-tcp-port-is-available

Net-net: it's laborious and requires iteration over local endpoints.  Also, if something attaches to the port between the time of starting the iteration and the end, the iteration won't see it, which will cause your code to think it's available.  Best IMO to simply catch the exception and handle it accordingly.

Cheers, Joel
